src/ctelnet.cpp

```diff
@@ -1413,6 +1467,72 @@ void cTelnet::processTelnetCommand(const std::string& command)
     case TN_SB: {
         option = command[2];
 
+        // CHARSET
+        if (option == OPT_CHARSET && enableCHARSET) {
+            QByteArray payload = command.c_str();
+            if (payload.size() < 6) {
+                return;
+            }
+
+            // Trim off the Telnet suboption bytes from beginning (3) and end (2)
+            payload = payload.mid(3, static_cast<int>(payload.size()) - 5);
+
+            // CHARSET support per https://tools.ietf.org/html/rfc2066
+            if (command[3] == CHARSET_REQUEST) {
+                if (payload.startsWith("[TTABLE]1")) { // No translate table support.  Discard.
+                    payload.remove(0, 9);
+                }
+
+                QList<QByteArray> characterSetList = payload.split(payload[1]); // Second character is the separator.
```

Comment:
:information_source: There is a `QByteArrayList` typedef that can be used here - but since we are in the age of C++17 we can forget about explicitly saying what the type here is and just use `auto` to let the compiler handle it...